### PR TITLE
feat: add useTheme hook for automatic dark mode detection

### PR DESCRIPTION
**Pull request by KaynanWallen**
## Summary by CodeRabbit

- **New Features**
  - Added a React hook that auto-syncs the UI theme with the system dark/light preference and exposed it in the public API.

- **Tests**
  - Added comprehensive tests covering initialization, theme application, media-query listener behavior, cleanup, and edge cases.

- **Chores**
  - Bumped package version to 1.1.49.
  - Updated build configuration to include the new hook in the distributed package.

### CODE REVIEW

File: src/hooks/useTheme.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        htmlElement.setAttribute('data-theme', originalTheme);
+      }
+    };
+
+    // Aplica o theme inicial
+    applyTheme();
+
+    // Monitora mudanças nas preferências do sistema
+    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+    const handleChange = (_event: MediaQueryListEvent) => {
```
Comment (souzajr) [inline src/hooks/useTheme.ts:39]: se não vai usar o event melhor remover

**Review by souzajr — CHANGES_REQUESTED** [review]

**Review by souzajr — APPROVED** [review]
